src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -6,20 +6,34 @@
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class ReadWriteList<E> {
-    private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final List<E> list = new ArrayList<>();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
         // write your code here
+        try {
+            lock.writeLock().lock();
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        try {
+            lock.readLock().lock();
+            return list.get(index);
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public int size() {
```

Comment:
While the 'size' method is correctly implemented, if the size of the list is not expected to change frequently, you might consider caching the size to avoid acquiring the read lock every time 'size' is called. This could improve performance in a read-heavy environment.